nixos/modules/services/networking/prosody.nix

```diff
@@ -422,6 +422,13 @@ in
         description = "List of administrators of the current host";
       };
 
+      authentication = mkOption {
+        type = types.str;
+        default = "internal_hashed";
+        example = "internal_plain";
```

Comment:
Is `internal_hashed` the default if not set?

And if there are few enough options, a `types.enum [ "internal_hashed" ... ]` would work well here.

Reply:
Prosody uses internal_plain as default value if no value is present in the config. Is there a convention to use the same default values the application provides?

Reply:
Not necessarily, but I'm worried about backwards compatibility

Reply:
As described in the commit message: Old installations will be changed to internal_hashed, if nothing else was specified in extraConfig. As described [here](https://prosody.im/doc/modules/mod_auth_internal_hashed#activating): "When you first activate hashed storage, user accounts will automatically be upgraded from plain to hashed one-by-one as they log in. "

Reply:
Ahh sorry, I was confused by it at first, thanks